Modules/Microsoft365DSC/DSCResources/MSFT_TeamsGroupPolicyAssignment/MSFT_TeamsGroupPolicyAssignment.psm1

```diff
@@ -0,0 +1,433 @@
+        if($null -eq $GroupPolicyAssignemnt)
+        {
+            Write-Verbose -Message "GroupPolicyAssignment not found for $GroupDisplayname"
+            return $nullReturn
+        }
+        Write-Verbose -Message "Found GroupPolicyAssignment $($Group.Displayname) with PolicyType:$($GroupPolicyAssignemnt.PolicyType) and Policy Name:$($GroupPolicyAssignemnt.PolicyName)"
+        return @{
+            GroupId               = $Group.Id
+            GroupDisplayname      = $Group.Displayname
+            PolicyType            = $GroupPolicyAssignemnt.PolicyType
+            PolicyName            = $GroupPolicyAssignemnt.PolicyName
+            Priority              = $GroupPolicyAssignemnt.Priority
+            Ensure                = 'Present'
+            Credential            = $Credential
+            ApplicationId         = $ApplicationId
+            TenantId              = $TenantId
+            CertificateThumbprint = $CertificateThumbprint
+        }
+    }
+    catch
+    {
+        New-M365DSCLogEntry -Message 'Error retrieving data:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return $nullReturn
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Id,
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $GroupDisplayname,
+
+        [Parameter()]
+        [System.String]
+        $GroupId,
+
+        [Parameter()]
+        [System.String]
+        $PolicyType,
+
+        [Parameter()]
+        [System.String]
+        $PolicyName,
+
+        [Parameter()]
+        [System.String]
+        $Priority,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint
+    )
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftTeams' -InboundParameters $PSBoundParameters
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    #check policyname
+    $command = "get-cs" + $PolicyType
```

Comment:
What are you trying to do here? According to the documentation, the PolicyType parameter can be any of these values:

- CallingLineIdentity
- TeamsAppSetupPolicy
- TeamsAudioConferencingPolicy
- TeamsCallingPolicy
- TeamsCallParkPolicy
- TeamsChannelsPolicy
- TeamsComplianceRecordingPolicy
- TenantDialPlan
- TeamsMeetingBroadcastPolicy
- TeamsMeetingPolicy
- TeamsMessagingPolicy
- TeamsShiftsPolicy
- TeamsUpdateManagementPolicy
- TeamsVerticalPackagePolicy
Source: https://learn.microsoft.com/en-us/powershell/module/teams/new-csgrouppolicyassignment?view=teams-ps#-policytype

For several of these values, there is no Get-Cs..... cmdlet. Like the first two: CallingLineIdentity and TeamsAppSetupPolicy. I think this can result in errors, especially since you are not doing any validation and directly inserting the command into Invoke-Expression. This can lead to security issues, where someone specifies "whatever -ErrorAction SilentlyContinue;Remove-Item -Path C:\ -Recurse" into the PolicyType parameter.